### ISSUE TITLE
Remove `dataObjLock` and `dataObjUnlock` APIs

### ISSUE THREAD

**Issue by korydraughn** [irods/irods#7980]
- [x] main only

---

Depends on #7979

**Comment by alanking** [comment]
We may be able to remove `fsDataObjLock` and `fsDataObjUnlock` as well:
https://github.com/irods/irods/blob/040335a57f04d347c119d78b4d1f8977f5d88bf5/server/core/include/irods/physPath.hpp#L107-L109

While `--rlock` and `--wlock` are not deprecated on the iCommands, I think removal of these APIs makes them completely ineffective. We may have to leave them in since they weren't deprecated, though.

**Comment by korydraughn** [comment]
I think it's safe to deprecate them for 4.3.5 and go ahead and remove them in 5.0.0 since their on separate timelines.

**Comment by alanking** [comment]
Okay. Should we make a separate issue for removing `--rlock` and `--wlock` since they are more specific to the iCommands, or just bundle them in with this issue? We should definitely make a separate issue for deprecating them for 4.3.5.

**Comment by korydraughn** [comment]
Separate issues sound good.

**Comment by alanking** [comment]
Created https://github.com/irods/irods/issues/8272 and https://github.com/irods/irods/issues/8273 for deprecation and removal of `--wlock`/`--rlock`.
